custom/src/main/java/com/solarwinds/opentelemetry/extensions/config/DeclarativeLoader.java

```diff
@@ -0,0 +1,65 @@
+package com.solarwinds.opentelemetry.extensions.config;
+
+import com.google.auto.service.AutoService;
+import com.solarwinds.joboe.config.ConfigContainer;
+import com.solarwinds.joboe.config.ConfigGroup;
+import com.solarwinds.joboe.config.InvalidConfigException;
+import com.solarwinds.joboe.config.JavaRuntimeVersionChecker;
+import com.solarwinds.joboe.logging.Logger;
+import com.solarwinds.joboe.logging.LoggerFactory;
+import com.solarwinds.opentelemetry.extensions.LoggingConfigProvider;
+import com.solarwinds.opentelemetry.extensions.config.parser.yaml.DeclarativeConfigParser;
+import com.solarwinds.opentelemetry.extensions.provider.AutoConfigurationCustomizerProviderImpl;
+import io.opentelemetry.api.incubator.config.ConfigProvider;
+import io.opentelemetry.api.incubator.config.DeclarativeConfigProperties;
+import io.opentelemetry.javaagent.tooling.BeforeAgentListener;
+import io.opentelemetry.sdk.autoconfigure.AutoConfiguredOpenTelemetrySdk;
+import io.opentelemetry.sdk.autoconfigure.internal.AutoConfigureUtil;
+
+@AutoService(BeforeAgentListener.class)
+public class DeclarativeLoader implements BeforeAgentListener {
+  private static final Logger logger = LoggerFactory.getLogger();
+
+  @Override
+  public void beforeAgent(AutoConfiguredOpenTelemetrySdk autoConfiguredOpenTelemetrySdk) {
+    ConfigProvider configProvider =
+        AutoConfigureUtil.getConfigProvider(autoConfiguredOpenTelemetrySdk);
+
+    if (configProvider != null) {
+      boolean jdkVersionSupported = JavaRuntimeVersionChecker.isJdkVersionSupported();
+      AutoConfigurationCustomizerProviderImpl.setAgentEnabled(jdkVersionSupported);
+
```

Comment:
It seems the declarative config handling below is run regardless JDK version supported / agent enabled... is that intentional and if so, why?

Reply:
no, it wasn't. also is this restriction still valid, i'm not remembering why we have it?